apps/hermes/client/js/src/HermesConnection.ts

```diff
@@ -0,0 +1,222 @@
+   * This will throw an error if there is a network problem or the price service returns a non-ok response.
+   *
+   * @param publishTime Unix timestamp in seconds.
+   * @param ids Array of hex-encoded price feed IDs for which updates are requested.
+   * @param options Optional parameters:
+   *        - encoding: Encoding type. If specified, return the price update in the encoding specified by the encoding parameter. Default is hex.
+   *        - parsed: Boolean to specify if the parsed price update should be included in the response. Default is false.
+   *
+   * @returns Array of PriceUpdate objects containing the updates at the specified timestamp.
+   */
+  async getPriceUpdatesAtTimestamp(
+    publishTime: UnixTimestamp,
+    ids: HexString[],
+    options?: {
+      encoding?: EncodingType;
+      parsed?: boolean;
+    }
+  ): Promise<PriceUpdate[]> {
+    const url = new URL(`${this.baseURL}/v2/updates/price/${publishTime}`);
+    for (const id of ids) {
+      url.searchParams.append("ids[]", id);
+    }
+
+    if (options) {
+      this.appendUrlSearchParams(url, options);
+    }
+
+    return await this.httpRequest<PriceUpdate[]>(url.toString());
+  }
+
+  /**
+   * Fetch streaming price updates for a set of price feed IDs.
+   * This endpoint can be customized by specifying the encoding type, whether the results should include parsed updates,
+   * and if unordered updates or only benchmark updates are allowed.
+   * This will return an EventSource that can be used to listen to streaming updates.
+   * If an invalid hex-encoded ID is passed, it will throw an error.
+   *
+   *
+   * @param ids Array of hex-encoded price feed IDs for which streaming updates are requested.
+   * @param encoding Optional encoding type. If specified, updates are returned in the specified encoding. Default is hex.
+   * @param parsed Optional boolean to specify if the parsed price update should be included in the response. Default is false.
+   * @param allow_unordered Optional boolean to specify if unordered updates are allowed to be included in the stream. Default is false.
+   * @param benchmarks_only Optional boolean to specify if only benchmark prices that are the initial price updates at a given timestamp (i.e., prevPubTime != pubTime) should be returned. Default is false.
+   * @returns An EventSource instance for receiving streaming updates.
+   */
+  async getPriceUpdatesStream(
+    ids: HexString[],
+    options?: {
+      encoding?: EncodingType;
+      parsed?: boolean;
+      allow_unordered?: boolean;
+      benchmarks_only?: boolean;
+    }
+  ): Promise<EventSource> {
+    const url = new URL("/v2/updates/price/stream", this.baseURL);
+    ids.forEach((id) => {
+      url.searchParams.append("ids[]", id);
+    });
+
+    if (options) {
+      this.appendUrlSearchParams(url, options);
+    }
+
+    return new EventSource(url.toString());
+  }
+
+  private appendUrlSearchParams(url: URL, params: Record<string, any>) {
```

Comment:
You should not use `any` here, it's very unsafe.  In general you shouldn't use `any` _ever_, it effectively bails out of the type system completely.  You can use `unknown` if you want to allow any type -- and this will ensure typescript can make sure that your code is safe for an arbitrary type -- but in this case I'd just type it as `EncodingType | boolean` since that's all you've tested it to work for anyways.

Reply:
thanks for the tip!